### PR TITLE
Fix open pull request against the wrong branch

### PR DESCRIPTION
**Pull request by omerzi**
- [ ] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
---

### CODE REVIEW

File: commands/utils/params.go [hunk 1]
```diff
@@ -389,7 +393,9 @@ func extractGitInfoFromEnvs() (*GitClientInfo, error) {
 	if err = readParamFromEnv(GitBaseBranchEnv, &branch); err != nil && !e.IsMissingEnvErr(err) {
 		return nil, err
 	}
-	clientInfo.Branches = []string{branch}
+	if branch != "" {
+		clientInfo.Branches = []string{branch}
```
Comment (sverdlov93) [inline commands/utils/params.go:397]: why not clientInfo.Branch (string) ?

File: commands/utils/git.go [hunk 2]
```diff
@@ -152,6 +151,14 @@ func (gm *GitManager) CreateBranchAndCheckout(branchName string) error {
 }
 
 func (gm *GitManager) createBranchAndCheckout(branchName string, create bool) error {
+	currentBranch, err := getCurrentBranch(gm.repository)
```
Comment (sverdlov93) [inline commands/utils/git.go:154]: i think maybe you can skip this check an just let Git to run the checkout and I believe that they do this check

File: commands/utils/params.go [hunk 3]
```diff
@@ -180,36 +185,40 @@ func (jp *JFrogPlatform) setDefaultsIfNeeded() (err error) {
...
-		if gitClientInfo.RepoName == "" {
+		if gitParamsFromEnv.RepoName == "" {
 			return fmt.Errorf("repository name is missing. please set the repository name in your %s file or as the %s environment variable", FrogbotConfigFile, GitRepoEnv)
 		}
-		g.RepoName = gitClientInfo.RepoName
+		g.RepoName = gitParamsFromEnv.RepoName
 	}
 	if len(g.Branches) == 0 {
-		g.Branches = append(g.Branches, gitClientInfo.Branches...)
+		if len(gitParamsFromEnv.Branches) == 0 {
```
Comment (sverdlov93) [inline commands/utils/params.go:213]: g.Branches = gitParamsFromEnv.Branches
if len(g.Branches) == 0 {
var checkoutBrunch string
			if checkoutBrunch, err = GetCheckoutBranchFromDotGit(); err != nil {
				return
			}
g.Branches = []string{checkoutBrunch}
}

Comment (omerzi) [reply commands/utils/params.go:213]: @sverdlov93, the len check is a validation that the branches weren't received from the frogbot-config.yml file

**Pull request merged**